sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemPropertiesTable.java

```diff
@@ -0,0 +1,191 @@
+        "authenticationResult in dataContext"
+    );
+    SystemSchema.checkStateReadAccessForServers(authenticationResult, authorizerMapper);
+    final Iterator<DiscoveryDruidNode> druidServers = SystemSchema.getDruidServers(druidNodeDiscoveryProvider);
+
+    final Map<String, ServerProperties> serverToPropertiesMap = new HashMap<>();
+    druidServers.forEachRemaining(discoveryDruidNode -> {
+      final DruidNode druidNode = discoveryDruidNode.getDruidNode();
+      final Map<String, String> propertiesMap = getProperties(druidNode);
+      if (serverToPropertiesMap.containsKey(druidNode.getHostAndPortToUse())) {
+        ServerProperties serverProperties = serverToPropertiesMap.get(druidNode.getHostAndPortToUse());
+        serverProperties.addNodeRole(discoveryDruidNode.getNodeRole().getJsonName());
+      } else {
+        serverToPropertiesMap.put(druidNode.getHostAndPortToUse(), new ServerProperties(druidNode.getServiceName(), druidNode.getHostAndPortToUse(), new ArrayList<>(Arrays.asList(discoveryDruidNode.getNodeRole().getJsonName())), propertiesMap));
+      }
+    });
+    return Linq4j.asEnumerable(serverToPropertiesMap.values().stream().flatMap(ServerProperties::toRows).collect(Collectors.toList()));
+  }
+
+  private Map<String, String> getProperties(DruidNode druidNode)
+  {
+    final String url = druidNode.getUriToUse().resolve("/status/properties").toString();
+    try {
+      final Request request = new Request(HttpMethod.GET, new URL(url));
+      final StringFullResponseHolder response;
+      response = httpClient
+          .go(request, new StringFullResponseHandler(StandardCharsets.UTF_8))
+          .get();
+
+      if (response.getStatus().getCode() != HttpServletResponse.SC_OK) {
+        throw new RE(
+            "Failed to get properties from node[%s]. Error code[%d], description[%s].",
+            url,
+            response.getStatus().getCode(),
+            response.getStatus().getReasonPhrase()
+        );
+      }
+      return jsonMapper.readValue(
+          response.getContent(),
+          new TypeReference<Map<String, String>>(){}
+      );
+    }
+    catch (Exception e) {
+      throw DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.UNCATEGORIZED)
+                          .build(e, "HTTP request to[%s] failed", url);
+    }
+  }
+
+  private static class ServerProperties
+  {
+    final String serviceName;
+    final String server;
+    final List<String> nodeRoles;
+    final Map<String, String> properties;
+
+    public ServerProperties(String serviceName, String server, List<String> nodeRoles, Map<String, String> properties)
+    {
+      this.serviceName = serviceName;
+      this.server = server;
+      this.nodeRoles = nodeRoles;
+      this.properties = properties;
+    }
+
+    public void addNodeRole(String nodeRole)
+    {
+      nodeRoles.add(nodeRole);
+    }
+
+    public Stream<Object[]> toRows()
```

Comment:
this method should return a `List<Object[]>` rows since we are going to always convert it to a list anyway.